Clock and stopper classes + tests for each one of them (not all tests covered)

Afek-Sakaju opened the pull request.

Commit:
clock class finished

Commit:
clock class optimized and finished
just remaining to finish tests writing here

Commit:
stopper and clock classes finished
remained both tests writing

Commit:
13 tests written to stopper & clock

hdriel commented on classes/clock/clock.js:
```diff
@@ -27,7 +27,7 @@ class Clock extends Time {
 
         this.interval = setInterval(() => {
-            if (this.totalSeconds === MAX_CLOCK_SECONDS) this.pause();
+            if (this.tSeconds === MAX_CLOCK_SECONDS) this.pause();
```
I prefer first to add seconds 
then check if this.tSeconds >= MAX_CLOCK_SECONDS if so, pause 

because the clock can be not valid after to added one seconds last

Afek-Sakaju replied:
statement changed to stop clock as it reaches limit.
adding 1 second happens before the statement so the limit, wont break

hdriel commented on classes/clock/__tests__/clock.test.js:
```diff
@@ -0,0 +1,97 @@
+                const maxLimit = '23:59:59';
+
+                setTimeout(() => {
+                    expect(clock.toString()).toBe(maxLimit);
```
this line can't be reached 
please try to use it in another way 
try to use done param from test like 

```
test('clock: 23:59:50 auto pauses after reaching "23:59:59"', (done) => {
setTimeout(() => {
       expect(clock.toString()).toBe(maxLimit);
       done()
 }, 30000);
}
```

Afek-Sakaju replied:
hadriel after i checked again, this line reached in the code
because i added before all tests:
`jest.useFakeTimers();`

and in each test the code: 
`jest.runAllTimers();`

hdriel commented on classes/clock/__tests__/clock.test.js:
```diff
@@ -0,0 +1,97 @@
+            
+            })*/
+
+            jest.useFakeTimers();
```
I dont know about jest.useFakeTimers and jest.runAllTimers
I would happy to learn about that from you 😊

Afek-Sakaju replied:
**Ty boss, its a big respect for me to teach you**

hdriel commented on classes/stopper/__tests__/stopper.test.js:
```diff
@@ -0,0 +1,100 @@
+                [{ seconds: 10, minutes: 30, hours: 1 }, '01:30:10'],
+            ])(
+                'stopper of %s units, autoStart set to false then after 15 seconds returns %s',
+                (params, result) => {
```
I've not tried to use done param with test.each , 
you are welcome to explore that and update me :)

Afek-Sakaju replied:
ok ill check that sir

hdriel commented on classes/stopper/stopper.js:
```diff
@@ -0,0 +1,48 @@
+        clearInterval(this.interval);
+    }
+
+    say() {
```
what is for?

hdriel commented on classes/stopper/stopper.js:
```diff
@@ -0,0 +1,48 @@
+
+        this.interval = setInterval(() => {
+            if (this.tSeconds === MIN_STOPPER_SECONDS) this.pause();
+            else this.subSeconds(1);
```
same comment as clock class as comment

hdriel commented on classes/stopper/stopper.js:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    validateLimiter() {
+        if (this.tSeconds > MAX_STOPPER_SECONDS) {
```
why did you change the `this.totalSeconds` to `this.tSeconds` ?

Afek-Sakaju replied:
because there is a get method at the Time class called totalSeconds

hdriel commented on classes/stopper/stopper.js:
```diff
@@ -0,0 +1,48 @@
+
+// check what's the time limits of stopper?
+
+class Stopper extends Time {
```
in the stopper class, should have also stop method , 
todo: implement stop method

hdriel commented on classes/clock/clock.js:
```diff
@@ -27,7 +27,7 @@ class Clock extends Time {
         this.interval = setInterval(() => {
-            if (this.totalSeconds === MAX_CLOCK_SECONDS) this.pause();
+            if (this.tSeconds === MAX_CLOCK_SECONDS) this.pause();
             else this.addSeconds(1);
```
I recommend to use `super.addSeconds(1)` although the **this** will work as well 
I think it's nicer to see where this method came from the inheritance..

afekTheMiniLearner replied:
so should i always use super in this cases?

Afek-Sakaju replied:
changed

Commit:
.

Commit:
both classes used methods
that came by inheritance
then i changed calling them by
"this" to "super" for more understandable code

Commit:
stop method created+
tests
